test/compiler/inference.jl

```diff
@@ -1562,6 +1562,12 @@ end
 @test arraysize_tfunc(Vector, Float64) === Union{}
 @test arraysize_tfunc(String, Int) === Union{}
 
+# tuple_tfunc
+# https://github.com/JuliaLang/julia/issues/44705
+import Core.Compiler: tuple_tfunc, widenconst
+@test widenconst(tuple_tfunc(Any[Type{Int}])) === Tuple{DataType}
+@test tuple_tfunc(Any[Union{Type{Int32},Type{Int64}}]) === Tuple{Type}
```

Comment:
Why not `Tuple{DataType}`?

Reply:
We would need a more precise idea of widenconst that does widenconcrete, and we don't have that right now